crates/bevy_math/src/aspect_ratio.rs

```diff
@@ -11,17 +11,60 @@ use bevy_reflect::Reflect;
 pub struct AspectRatio(f32);
 
 impl AspectRatio {
-    /// Create a new `AspectRatio` from a given `width` and `height`.
+    /// Standard 16:9 aspect ratio
+    pub const SIXTEEN_NINE: Self = Self(16.0 / 9.0);
+    /// Standard 4:3 aspect ratio
+    pub const FOUR_THREE: Self = Self(4.0 / 3.0);
+    /// Standard 21:9 ultrawide aspect ratio
+    pub const ULTRAWIDE: Self = Self(21.0 / 9.0);
+
+    /// Create a new ``AspectRatio`` from a given width and height.
     #[inline]
     pub fn new(width: f32, height: f32) -> Self {
         Self(width / height)
     }
 
-    /// Create a new `AspectRatio` from a given amount of `x` pixels and `y` pixels.
+    /// Create a new ``AspectRatio`` from a given amount of x pixels and y pixels.
     #[inline]
     pub fn from_pixels(x: u32, y: u32) -> Self {
         Self::new(x as f32, y as f32)
     }
+
+    /// Create a new ``AspectRatio`` from a given width and height.
+    /// Returns None if height is zero or if the result is not finite.
+    #[inline]
+    pub fn try_new(width: f32, height: f32) -> Option<Self> {
+        let ratio = width / height;
+        if ratio.is_finite() && height != 0.0 {
+            Some(Self(ratio))
+        } else {
+            None
+        }
+    }
+
+    /// Returns the inverse of this aspect ratio (height/width).
+    #[inline]
+    pub fn inverse(&self) -> Self {
+        Self(1.0 / self.0)
+    }
+
+    /// Returns true if the aspect ratio represents a landscape orientation.
+    #[inline]
+    pub fn is_landscape(&self) -> bool {
+        self.0 > 1.0
+    }
+
+    /// Returns true if the aspect ratio represents a portrait orientation.
+    #[inline]
+    pub fn is_portrait(&self) -> bool {
+        self.0 < 1.0
+    }
+
+    /// Returns true if the aspect ratio is exactly square.
+    #[inline]
+    pub fn is_square(&self) -> bool {
+        (self.0 - 1.0).abs() < f32::EPSILON
```

Comment:
Why is this more complex than the other ones? landscape and Portrait don't acknowledge for negative numbers but here we do? Also then there's the EPSILON comparison, so can it happen that one aspect ratio is both landscape and square?

Reply:
You're right, this implementation is more complex than the others. The added complexity in the `is_square` method comes from an attempt to handle floating-point precision issues. When dealing with floating-point calculations, exact equality comparisons can be problematic due to rounding errors. The `EPSILON` comparison was introduced to allow for small floating-point discrepancies, making the `is_square` check more robust.

However, I acknowledge that this approach creates inconsistency with the simpler implementations of `is_landscape` and `is_portrait`. Thank you for pointing this out. We should consider whether this level of precision is necessary for our use case, or if we can simplify the implementation to be more consistent with the other methods.

I completely agree that the current code lacks consistency and could lead to logical contradictions.

To address these concerns, I propose the following changes:

1. Introduce a consistent use of EPSILON across all three methods:

```rust
impl AspectRatio {
    const EPSILON: f32 = 1e-6;

    pub fn is_landscape(&self) -> bool {
        self.0 > 1.0 + Self::EPSILON
    }

    pub fn is_portrait(&self) -> bool {
        self.0 < 1.0 - Self::EPSILON
    }

    pub fn is_square(&self) -> bool {
        (self.0 - 1.0).abs() <= Self::EPSILON
    }
}
```

2. Enhance the constructor to ensure valid inputs:

```rust
impl AspectRatio {
    pub fn new(width: f32, height: f32) -> Result<Self, &'static str> {
        if width <= 0.0 || height <= 0.0 {
            Err("Width and height must be positive")
        } else if !width.is_finite() || !height.is_finite() {
            Err("Width and height must be finite")
        } else {
            Ok(Self(width / height))
        }
    }
}
```

What do you think about these proposed changes?

Reply:
You really did not just give me a ChatGPT answer did you? ._.

Reply:
The code wasn't used, but the translation was done with Claude and DeepL.

Reply:
I will try to avoid use a translator.
I didn't know it would be a problem, sorry.

Reply:
> I will try to avoid use a translator. I didn't know it would be a problem, sorry.

I wouldn't worry too much about it. Just be aware that there have been cases of people trying to push AI generated changes to the project (and in other projects too) so people are naturally a bit cautious when they see something that looks AI generated. Even when used as a translator, these tools have a tendency to output AI-isms that look weird to native speakers.

Just don't feel the need or pressure to use perfect english. Code review is meant to help you with getting things right anyway.

Reply:
I checked! Thanks to your advice.